avocado/query/utils.py

```diff
@@ -0,0 +1,138 @@
+import logging
+from django.db import connections, DEFAULT_DB_ALIAS
+from django.core.cache import cache
+
+logger = logging.getLogger(__name__)
+
+
+TEMP_DB_ALIAS_PREFIX = '_db:{0}'
+
+
+def named_connection(name, db=DEFAULT_DB_ALIAS):
+    """Opens a new and isolated connection for the named queryset.
+
+    The name is used to create a cache key for storing the backend connection
+    ID which enables it to be canceled later.
+
+    The implementation of this is a tad conflated due to how Django manages
+    connections.
+    """
+    # Define a new database alias
+    temp_db = TEMP_DB_ALIAS_PREFIX.format(name)
+
+    if temp_db in connections.databases:
+        conn = connections[temp_db]
```

Comment:
I'm a bit confused by this line. Does indexing `connections` actually index into `connections.databases`? In the line above, you check if `temp_db` is in `connections.databases` but here, you are using `temp_db` to get an item directly from `connections`. Might be worth a comment to clear up why the key check and get item are done on two different collections.